src/java/org/apache/cassandra/db/compression/ZstdDictionaryTrainer.java

```diff
@@ -68,17 +68,23 @@ public class ZstdDictionaryTrainer implements ICompressionDictionaryTrainer
     private volatile TrainingStatus currentTrainingStatus;
     private volatile String failureMessage;
 
-    public ZstdDictionaryTrainer(String keyspaceName, String tableName,
-                                 CompressionDictionaryTrainingConfig config,
-                                 int compressionLevel)
+    public ZstdDictionaryTrainer(String keyspaceName, String tableName, int compressionLevel)
+    {
+        this(keyspaceName,
+             tableName,
+             compressionLevel,
+             Math.round(1 / DatabaseDescriptor.getCompressionDictionaryTrainingSamplingRate()));
+    }
+
+    @VisibleForTesting
+    public ZstdDictionaryTrainer(String keyspaceName, String tableName, int compressionLevel, int samplingRate)
```

Comment:
You are passing `1/samplingRate` here, should the name of the variable be percentage? or any other appropriate name
```suggestion
    public ZstdDictionaryTrainer(String keyspaceName, String tableName, int compressionLevel, int samplingPercentage)
```

Reply:
I would keep it as it is.